package/MDAnalysis/coordinates/XYZ.py

```diff
@@ -302,7 +302,11 @@ def __init__(self, filename, **kwargs):
         # the filename has been parsed to be either be foo.xyz or foo.xyz.bz2 by
         # coordinates::core.py so the last file extension will tell us if it is
         # bzipped or not
-        root, ext = os.path.splitext(self.filename)
+        if util.isstream(self.filename):
+            root = str(self.filename)
```

Comment:
`root` in the old and new code are not the same. It might not matter here. 

However, a cleaner way would be to have our own `splitext` that works on streams and use it here.

The best solution would be to think about `NamedStream` and see what makes sense with Python 3.